### PR TITLE
Updated the generic Deployment article

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.3+ |
| Fixed tickets | - |

### CODE REVIEW

File: cookbook/deployment/tools.rst [hunk 1]
```diff
@@ -8,9 +8,9 @@ How to Deploy a Symfony Application
 
 .. note::
 
-    Deploying can be a complex and varied task depending on your setup and needs.
-    This entry doesn't try to explain everything, but rather offers the most
-    common requirements and ideas for deployment.
+    Deploying can be a complex and varied task depending on the setup and
+    requirements of your application. This article is not a step-by-step guide,
```
Comment (xabbuh) [inline cookbook/deployment/tools.rst]: [...] and the requirements [...]

File: cookbook/deployment/tools.rst [hunk 2]
```diff
@@ -8,9 +8,9 @@ How to Deploy a Symfony Application
...
+    but a general list of the most common requirements and ideas for deployment.
```
Comment (xabbuh) [inline cookbook/deployment/tools.rst]: [...] but is a [...]

File: cookbook/deployment/tools.rst [hunk 3]
```diff
@@ -19,26 +19,27 @@ Symfony Deployment Basics
...
-A deployment may also include other things, such as:
+A deployment may also include other tasks, such as:
 
-* Tagging a particular version of your code as a release in your source control repository;
+* Tagging a particular version of your code as a release in your source control
+  repository;
 * Creating a temporary staging area to build your updated setup "offline";
 * Running any tests available to ensure code and/or server stability;
-* Removal of any unnecessary files from ``web`` to keep your production environment clean;
+* Removal of any unnecessary files from ``web/`` directory to keep your production
```
Comment (xabbuh) [inline cookbook/deployment/tools.rst]: [...] from the `web/` directory [...]

**Comment by javiereguiluz** [comment]
@xabbuh thanks for your review. Everything is fixed now.

File: cookbook/deployment/tools.rst [hunk 4]
```diff
@@ -19,26 +19,27 @@ Symfony Deployment Basics
 
 The typical steps taken while deploying a Symfony application include:
 
-#. Upload your modified code to the live server;
-#. Update your vendor dependencies (typically done via Composer, and may
+#. Upload your code to the production server;
+#. Install/update your vendor dependencies (typically done via Composer, and may
```
Comment (WouterJ) [inline cookbook/deployment/tools.rst]: you should remove the serial comma before "and"

Comment (WouterJ) [inline cookbook/deployment/tools.rst]: during a deploy, I would only advertise to install deps.

File: cookbook/deployment/tools.rst [hunk 5]
```diff
@@ -62,12 +63,35 @@ manually taking other steps (see `Common Post-Deployment Tasks`_).
...
+    This Capistrano-like deployment tool is built in PHP, and may be easier
+    for PHP developers to extend for their needs.
 
-See `The Tools`_ for a list of tools that can help with deployment.
+Bundles
+    There are some `bundles that add deployment features`_ directly into your
+    Symfony console.
+
+Basic scripting
+    You can of course use shell, `Ant`_, or any other build tool to script
```
Comment (WouterJ) [inline cookbook/deployment/tools.rst]: again, you should remove the serial comma here

File: cookbook/deployment/tools.rst [hunk 6]
```diff
@@ -163,51 +185,17 @@ Don't forget that deploying your application also involves updating any dependen
...
-    PaaS is a relatively new way to deploy your application. Typically a PaaS
-    will use a single configuration file in your project's root directory to
-    determine how to build an environment on the fly that supports your software.
-    One provider with confirmed Symfony support is `PagodaBox`_.
-
 .. tip::
 
-    Looking for more? Talk to the community on the `Symfony IRC channel`_ #symfony
-    (on freenode) for more information.
+    Looking for more? Talk to the community on the `Symfony IRC channel`_ for
```
Comment (WouterJ) [inline cookbook/deployment/tools.rst]: oeh, being part of the IRC group, I'm -1 on this one. These kind of messages send people with a wrong expectation to the IRC channel. I'm almost sure none of the "IRC guys" knows a lot more about deployment than what's written here.

**Comment by WouterJ** [comment]
Thanks, lots of good fixes in here! Please note that I've commented as if you added all these things. I know you didn't, but I'm too used to that way of reviewing. Take it as opinions on the content and not your diff :)

File: cookbook/deployment/tools.rst [hunk 7]
```diff
@@ -62,12 +63,35 @@ manually taking other steps (see `Common Post-Deployment Tasks`_).
...
+specifically tailored to the requirements of Symfony.
+
+`Capifony`_
+    Provides a specialized set of tools on top of `Capistrano`_, tailored
+    specifically to Symfony projects.
+
+`sf2debpkg`_
+    Helps you build a native Debian package for your Symfony project.
+
+`Magallanes`_
```
Comment (timglabisch) [inline cookbook/deployment/tools.rst]: can we please add fabric? :)

Comment (cordoval) [inline cookbook/deployment/tools.rst]: my favorite thanks!

Comment (javiereguiluz) [inline cookbook/deployment/tools.rst]: Thanks for your suggestion: Fabric is indeed very popular, so I've added a reference to it in 9e01941

**Comment by javiereguiluz** [comment]
No worries Wouter. I love your reviews! I've fixed everything you said and I've removed the reference to the IRC channel.

**Comment by weaverryan** [comment]
Nice improvements here :). Thanks as always Javier!

**Pull request merged**